### PR TITLE
[misc] promisify FileHandler and remove dependency on async

### PR DESCRIPTION
**Pull request by das7pad**
### Description
Followup for #60

This PR promisifies the FileHandler.

It also removes a now unused dependency `async`.

#### Related Issues / PRs
#60

#### Potential Impact
High, all requests run through this handler.

### CODE REVIEW

File: app/js/FileHandler.js [hunk 1]
```diff
@@ -1,176 +1,153 @@
...
 
-    const done = function(err, destPath) {
-      if (err) {
-        return callback(
-          new ConversionError({
-            message: 'error converting file',
-            info: { bucket, originalKey, opts }
-          }).withCause(err)
-        )
+  let promise
```
Comment (gh2k) [inline app/js/FileHandler.js]: We could simplify this further if this was `let destPath` and then we followed up with `destpath = await blah` in the conditional section. We'd then only need one `try`/`catch` block.

File: package.json [hunk 2]
```diff
@@ -21,7 +21,6 @@
   },
   "dependencies": {
     "@overleaf/o-error": "^2.1.0",
-    "async": "~0.2.10",
```
Comment (gh2k) [inline package.json]: :tada:

File: app/js/FileHandler.js (hunk 1 above)
Comment (das7pad) [reply app/js/FileHandler.js]: I chose to split this in regards to line 124 following -- we would rethrow this error with a different message from the catch block then.

Comment (gh2k) [reply app/js/FileHandler.js]: OK, fair enough. :slightly_smiling_face:

**Comment by das7pad** [comment]
Rebased to resolve conflicts.

**Pull request merged**